server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,635 @@
+            Arguments.of("scrambled content", Rule.Condition.Field.FROM, mimeMessageBuilder().addHeader("from", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULLNAME),
+            Arguments.of("folded content", Rule.Condition.Field.FROM, mimeMessageBuilder().addHeader("from", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+
+            Arguments.of("normal content", Rule.Condition.Field.TO, mimeMessageBuilder().addToRecipient(RECIPIENT_TO_1_FULL_ADDRESS), RECIPIENT_TO_1_USERNAME),
+            Arguments.of("multiple headers", Rule.Condition.Field.TO, mimeMessageBuilder().addToRecipient(RECIPIENT_TO_1_FULL_ADDRESS).addToRecipient(RECIPIENT_TO_2_FULL_ADDRESS), RECIPIENT_TO_2_USERNAME),
+            Arguments.of("scrambled content", Rule.Condition.Field.TO, mimeMessageBuilder().addHeader("to", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULLNAME),
+            Arguments.of("folded content", Rule.Condition.Field.TO, mimeMessageBuilder().addHeader("to", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+
+            Arguments.of("normal content", Rule.Condition.Field.CC, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS), "recipient_cc_1"),
+            Arguments.of("multiple headers", Rule.Condition.Field.CC, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS).addCcRecipient(RECIPIENT_CC_2_FULL_ADDRESS), "recipient_cc_2"),
+            Arguments.of("scrambled content", Rule.Condition.Field.CC, mimeMessageBuilder().addHeader("cc", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULLNAME),
+            Arguments.of("folded content", Rule.Condition.Field.CC, mimeMessageBuilder().addHeader("cc", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+
+            Arguments.of("normal content", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(SCRAMBLED_SUBJECT), "subject"),
+            Arguments.of("scrambled content", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(UNSCRAMBLED_SUBJECT), "subject Frédéric MARTIN")
+        );
+    }
+
+    @ParameterizedTest(name = "mailDirectiveShouldSetWhenContainsRuleValue when matching header field {1}, with {0}")
+    @MethodSource("mailDirectiveShouldSetWhenContainsRuleValueParamsProvider")
+    void mailDirectiveShouldSetWhenContainsRuleValue(String testDescription,
+                                                     Rule.Condition.Field fieldToMatch,
+                                                     MimeMessageBuilder mimeMessageBuilder,
+                                                     String valueToMatch,
+                                                     JMAPFilteringTestSystem testSystem) throws Exception {
+
+        testSystem.defineRuleForRecipient1(fieldToMatch, Rule.Condition.Comparator.CONTAINS, valueToMatch);
+
+        FakeMail mail = FakeMail.builder()
```

Comment:
The mail is always the same. Could we have a `asMail(MimeMessage m)` default method for  removing the boilerplate?